contracts/staking/interfaces/IPoSValidatorManager.sol

```diff
@@ -39,4 +113,43 @@ interface IPoSValidatorManager is IValidatorManager {
         bool includeUptimeProof,
         uint32 messageIndex
     ) external;
+
+    /**
+     * @notice Resubmits a delegator registration message to be sent to the P-Chain.
+     * Only necessary if the original message can't be delivered due to validator churn.
+     * @param validationID The ID of the validation period being registered.
+     * @param delegator The address of the delegator being registered.
+     */
+    function resendDelegatorRegistration(bytes32 validationID, address delegator) external;
+
+    /**
+     * @notice Completes the delegator registration process by returning an acknowledgement of the registration of a
+     * validationID from the P-Chain. After this function is called, the validator's weight is updated in the contract state.
+     * @param messageIndex The index of the Warp message to be received providing the acknowledgement.
+     * @param delegator The address of the delegator being registered.
+     */
+    function completeDelegatorRegistration(uint32 messageIndex, address delegator) external;
+
+    /**
+     * @notice Begins the process of removing a delegator from a validation period. The delegator must have been previously
+     * registered with the given validationID.
+     * @param validationID The ID of the validation period being removed.
+     */
+    function initializeEndDelegation(bytes32 validationID) external;
+
+    /**
+     * @notice Resubmits a delegator end message to be sent to the P-Chain.
+     * Only necessary if the original message can't be delivered due to validator churn.
+     * @param validationID The ID of the validation period being ended.
+     * @param delegator The address of the delegator being removed.
+     */
+    function resendEndDelegation(bytes32 validationID, address delegator) external;
```

Comment:
Technically our resends for starting or ending a delegation would only need to resend the current validator total weight with the highest nonce we've used so far. We don't need to store any of these messages, we just need to send our current state.

Reply:
I'm going to leave this PR storing the pending messages in the contract state, both for consistency, and because there's a potential gas tradeoff to reconstructing the Warp message when calling the `resend` methods. I'll create an issue to quantify that and proceed accordingly.

Reply:
I imagine the storage will cost more than the computation, and we should probably be optimizing for the happy path instead of the resend case. Happy to defer this to another ticket though

Reply:
Yup, don't want to make any assumptions about gas cost, so will defer that analysis and any implementation changes to a followup item.